test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -109,19 +120,62 @@ public void ChangePointDetectionWithSeasonality()
+                    0, -3.31410598754883, 0.5, 5.12000000000001E-08, 0, 1.5700820684432983, 5.2001145245395008E-07,
+                    0.012414560443710681, 0, 1.2854313254356384, 0.28810801662678009, 0.02038940454467935, 0,
+                    -1.0950627326965332, 0.36663890634019225, 0.026956459625565483
+                };
 
                 int index = 0;
                 while (enumerator.MoveNext() && index < expectedValues.Count)
                 {
                     row = enumerator.Current;
-                    Assert.Equal(expectedValues[index++], row.Change[0], precision: 7);  // Alert
-                    Assert.Equal(expectedValues[index++], row.Change[1], precision: 7);  // Raw score
-                    Assert.Equal(expectedValues[index++], row.Change[2], precision: 7);  // P-Value score
-                    Assert.Equal(expectedValues[index++], row.Change[3], precision: 7);  // Martingale score
+                    Assert.Equal(expectedValues[index++], row.Change[0], precision: 7); // Alert
+                    Assert.Equal(expectedValues[index++], row.Change[1], precision: 7); // Raw score
+                    Assert.Equal(expectedValues[index++], row.Change[2], precision: 7); // P-Value score
+                    Assert.Equal(expectedValues[index++], row.Change[3], precision: 7); // Martingale score
                 }
             }
         }
+
+        [Fact]
+        public void ChangePointDetectionWithSeasonalityPredictionEngine()
+        {
+            using (var env = new ConsoleEnvironment(conc: 1))
+            {
+                const int ChangeHistorySize = 10;
+                const int SeasonalitySize = 10;
+                const int NumberOfSeasonsInTraining = 5;
+                const int MaxTrainingSize = NumberOfSeasonsInTraining * SeasonalitySize;
+
+                List<Data> data = new List<Data>();
+                var dataView = env.CreateStreamingDataView(data);
+
+                var args = new SsaChangePointDetector.Arguments()
+                {
+                    Confidence = 95,
+                    Source = "Value",
+                    Name = "Change",
+                    ChangeHistoryLength = ChangeHistorySize,
+                    TrainingWindowSize = MaxTrainingSize,
+                    SeasonalWindowSize = SeasonalitySize
+                };
+
+                for (int j = 0; j < NumberOfSeasonsInTraining; j++)
+                for (int i = 0; i < SeasonalitySize; i++)
+                    data.Add(new Data(i));
+
+                for (int i = 0; i < ChangeHistorySize; i++)
+                    data.Add(new Data(i * 100));
+
+                // Train
+                SsaChangePointDetector detector = new SsaChangePointEstimator(env, args).Fit(dataView);
+                var engine = detector.MakeTimeSeriesPredictionFunction<Data, Prediction1>(env);
+                //Even though time series column is not requested it will pass the observation through time series transform.
```

Comment:
>//Even though time series column is not requested it will pass the observation through time series transform. [](start = 16, length = 109)

I wonder if this can be somehow checked. Probably not until you have saving implemented

Reply:
I added this check in another test.

---
In reply to: [233639726](https://github.com/dotnet/machinelearning/pull/1618#discussion_r233639726) [](ancestors = 233639726)